test_project/test_app/models.py

```diff
@@ -0,0 +1,55 @@
+from django.db import models
+from django.core.exceptions import ValidationError
+
+from .validators import validate_ingredients
+
+
+class Recipe(models.Model):
+    DRINK = 'drink'
+    MAIN_DISH = 'main_dish'
+    RECIPE_TYPES = (
+        (DRINK, 'Drink'),
+        (MAIN_DISH, 'Main Dish'),
+    )
+
+    name = models.CharField(max_length=25)
+    type = models.CharField(choices=RECIPE_TYPES, max_length=25)
+    ingredients = models.TextField(validators=[validate_ingredients])
+
+    def strip_empty_ingredients(self, ingredients):
```

Comment:
For example this method here, should it be part of the model class?
It breaks down the `ingredients` value, filters it and returns it. Does it need to be part of the model? Or can it simply be part of the `value` validation? That way the validation doesn't require a `Recipe` object to be instantiated.

Reply:
Take a look to my previous comment 😥